chain/chain/src/chain.rs

```diff
@@ -3181,147 +3176,545 @@ impl Chain {
         }
         Ok(res)
     }
-}
 
-/// Implement block merkle proof retrieval.
-impl Chain {
-    fn combine_maybe_hashes(
-        hash1: Option<MerkleHash>,
-        hash2: Option<MerkleHash>,
-    ) -> Option<MerkleHash> {
-        match (hash1, hash2) {
-            (Some(h1), Some(h2)) => Some(combine_hash(&h1, &h2)),
-            (Some(h1), None) => Some(h1),
-            (None, Some(_)) => {
-                debug_assert!(false, "Inconsistent state in merkle proof computation: left node is None but right node exists");
-                None
-            }
-            _ => None,
-        }
-    }
+    pub fn create_chunk_state_challenge(
```

Comment:
due to code moves -- the git shows this file as 2k diff - very had to see which parts of the functions where 'just' moved vs modified.